R/big_n.R

```diff
@@ -0,0 +1,104 @@
+#' Big N Structure
+#'
+#' Big N strucutre allows you to specify which values should become the big n
+#' values and how they should be formatted. Values are specified by providing
+#' the row value(s) of the parameter column for which the values are big n's.
+#' This will remove these from the body of the table and place them into columns
+#' matching the values in the column column(s). The default formatting is `N = xx`,
+#' on its own line, but that can be changed by providing a different `frmt()` to
+#' `n_frmt`
+#'
+#' @param param_val row value(s) of the parameter column for which the values
+#'   are big n's
+#' @param n_frmt [frmt()] to control the formatting of the big n's
+#'
+#' @return big_n_structure object
+#' @export
+#'
+big_n_structure<- function(param_val, n_frmt = frmt("\nN = xx")){
+  structure(
+    list(
+      param_val = param_val,
+      n_frmt = n_frmt
+    ),
+    class = c("big_n_structure","structure")
+  )
+}
+
+
+
+apply_big_n_df <- function(col_plan_vars, columns, value, big_n_df){
+  if(!is.null(big_n_df) && nrow(big_n_df) > 0){
+    col_lab <- columns %>% map_chr(as_label)
+    data_names <- col_plan_vars %>%
+      map_chr(as_label) %>%
+      split_data_names_to_df(data_names= c(), preselected_cols = .,
+                             column_names = col_lab)
+
+    for(i in seq(nrow(big_n_df))){
+      big_n_i <- big_n_df %>%
+        slice(i)
+      data_names <- data_names %>%
+        mutate(!!big_n_i$name := if_else(!!parse_expr(big_n_i$exp),
+                                         paste0(!!sym(big_n_i$name),
+                                                pull(big_n_i, !!value)),
+                                         !!sym(big_n_i$name)
+        ))
+    }
+
+    out <- unite_df_to_data_names(data_names, preselected_cols = c(), column_names = col_lab)
+  } else {
+    out <- col_plan_vars
+  }
+  out
+}
+
+remove_big_ns <- function(.data, param, big_n_structure){
+  if(!is.null(big_n_structure)){
+    .data <- .data %>%
+      filter(!(!!param) %in% big_n_structure$param_val)
+  }
+  .data
+}
+
+#' @importFrom dplyr slice_tail
+get_big_ns <-  function(.data, param, value, columns, big_n_structure, mock){
```

Comment:
for consistency's sake, should these be `values` and `column` to match `tfrmt()` args?